src/cpu_shader/flatten.rs

```diff
@@ -151,12 +311,176 @@ fn flatten_cubic(cubic: Cubic, line_ix: &mut usize, lines: &mut [LineSoup]) {
     }
 }
 
+fn flatten_arc(
+    path_ix: u32,
+    begin: Vec2,
+    end: Vec2,
+    center: Vec2,
+    angle: f32,
+    transform: &Transform,
+    line_ix: &mut usize,
+    lines: &mut [LineSoup],
+    bbox: &mut IntBbox,
+) {
+    let mut p0 = transform.apply(begin);
+    let mut r = begin - center;
+    let tol: f32 = 0.5;
+    let radius = tol.max((p0 - transform.apply(center)).length());
+    let x = 1. - tol / radius;
+    let theta = (2. * x * x - 1.).clamp(-1., 1.).acos();
```

Comment:
This is the same as sqrt(8 tol / radius) for small tol. I wonder if the extra math is worth it. I haven't investigated in detail.

Reply:
The new expression should be correct now and is simpler.